Max's Snake/Max's Snake/src/DynamicArray.c

```diff
@@ -0,0 +1,43 @@
+//
+//  DynamicArray.c
+//  Max's Snake
+//
+//  Created by Max Raulea on 06/04/2023.
+//
+
+
+#include "DynamicArray.h"
+#include <stdlib.h>
+
+const int MEM_BATCH_SIZE = 32;
+
+dArray* initArray(dArray *blocks){
+    if(!(blocks->block = malloc(MEM_BATCH_SIZE * sizeof(Block)))){
+        return NULL;
+    }
+    blocks->Size = MEM_BATCH_SIZE;
+    blocks->Lentgh = 1;
+    return blocks;
```

Comment:
Before returning, you should properly create a block.
Create a block same way you create a block inside the `AppendBlock()`